app/main.py

```diff
@@ -1 +1,30 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(self, name: str, health: int = 100) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = False
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}, "
+                f"Health: {self.health}, "
+                f"Hidden: {self.hidden}}}")
+
+    def die(self) -> None:
+        if self in Animal.alive:
+            Animal.alive.remove(self)
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    def bite(self, prey: Animal) -> None:
+        if isinstance(prey, Herbivore) and not prey.hidden:
+            prey.health -= 50
+            if prey.health <= 0:
+                prey.die()
```

Comment:
In the `bite` method, ensure that the prey is alive before attempting to bite. You can add a check like `if prey in Animal.alive:` before proceeding with the bite logic.